cfme/tests/automate/custom_button/test_infra_objects_ansible.py

```diff
@@ -0,0 +1,166 @@
+from collections import namedtuple
+
+import fauxfactory
+import pytest
+from widgetastic_patternfly import Dropdown
+
+from cfme import test_requirements
+from cfme.ansible import RemoteFile
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.tests.automate.custom_button import CredsHostsDialogView
+
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.conf import credentials
+
+pytestmark = [
+    pytest.mark.tier(2),
+    test_requirements.custom_button,
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE),
+]
+
+INFRA_OBJECTS = [
+    "PROVIDER",
```

Comment:
I think it would be better to use the ButtonGroupCollection class attributes directly here.

This can be a TODO/tech debt, but I think that we need an enum class for these type constants.

Reply:
@mshriver agreed with this point but Attributes covers all type of object. As you know we separated objects like infra, cloud, container and general. So for now i would like to TODO for me. Let me add one card. we will add changes in all custom button test files.